app/javascript/pages/components/DataViewer.jsx

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable max-len */
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/prop-types */
+import React from 'react';
+import axios from 'axios';
+import DatasetHeader from './partials/DatasetHeader';
+import DatasetTable from './partials/DatasetTable';
+
+export default class DataViewer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      availableYears: [],
+      table: '',
+      schema: '',
+      database: '',
+      queryYearColumn: '',
+      displayHeaders: [],
+      rowHeaders: [],
+      rows: [],
+      title: '',
+      source: '',
+      universe: '',
+      description: '',
+      metadata: {},
+      selectedYears: [],
+      currentPage: 1,
+    };
+    this.updateSelectedYears = this.updateSelectedYears.bind(this);
+    this.updatePage = this.updatePage.bind(this);
+  }
+
+  componentDidMount() {
+    const queryBase = 'https://prql.mapc.org/';
+    const queryToken = '16a2637ee33572e46f5609a578b035dc';
+    this.props.fetchDatasets().then((storeResponse) => {
+      const fullDataset = storeResponse.datasets.filter((dataset) => +dataset.seq_id === +this.props.match.params.id)[0];
+      const table = fullDataset.table_name;
```

Comment:
It might be worth considering [inline temporary variable](https://refactoring.guru/inline-temp) here. However if you don't want to write the entire thing later on, [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) could clean up all these lines as well.

Reply:
I went with the inline temp variable refactor here.